src/arviz_base/reorg.pyi

```diff
@@ -59,7 +59,7 @@ def explode_dataset_dims(
     labeller: Labeller | None = ...,
 ) -> xarray.Dataset: ...
 def references_to_dataset(
-    references: np.ScalarType,
+    references: int,
```

Comment:
The docstring is wrong, docstub doesn't allow commas to separate different types as a synonym of "or", so the 

```
references : scalar, 1D array-like, dict, DataArray, Dataset
```

docstring is not being parsed correctly, it should be something like

```
references : scalar | 1D array-like | dict | DataArray | Dataset
```

This is one of the cases where I think we want the docstring type to be overwriten by a more detailed type hint so that for example the keys and values of the dict case are also typed. There isn't much value in adding that directly in the type section of the docstring because it is covered in much more detail in the description, but it would be useful as type hint.

Reply:
but also, if we overwrite the docstring with more detailed type hint we could actually keep the docstring as:

```
references : scalar, 1D array-like, dict, DataArray or Dataset
```

future us might be confused about the syntax breaking elsewhere though, probably not worth it

Reply:
Doing `references : scalar or 1D array-like or dict or DataArray or Dataset`, I get:
```
def references_to_dataset(
    references: numbers.Number | ArrayLike | dict | xarray.DataArray | xarray.Dataset,
    ds: xarray.Dataset,
    sample_dims: Iterable[Hashable] | None = ...,
    ref_dim: str | list | None = ...,
) -> xarray.Dataset: ...
```
Isn't this a better option?

Reply:
Yes, I like it